core/commands/list_cloud_instances_command.rb

```diff
@@ -67,9 +69,30 @@ def generate_gcp_list
     end
     return Result.error('No instances were found in GCP-services') if all_instances.empty?
 
+    all_instances.each do |instance|
+      instance[:launch_time] = translate_to_right_time(instance[:launch_time])
+    end
+    select_by_time(all_instances) unless @env.hours.nil?
     Result.ok(all_instances)
   end
 
+  def translate_to_right_time(time)
+    date, times = time.split('T')
+    year, month, day = date.split('-')
+    hour, min, sec = times.split(':')
+    Time.utc(year, month, day, hour, min, sec) + 60 * 60 * 7 # Time delay for GCP-provider is 7 hours
+  end
+
+  def select_by_time(instances)
+    if @env.hours.to_i <= 0
+      @ui.error('The number of hours entered incorrectly. All instances will be shown.')
+      return instances
```

Comment:
This method modifies the passed parameter, do not do that. Create and use a new one instead.